<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>616</width>
    <height>271</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>616</width>
    <height>271</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>616</width>
    <height>271</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>RSA thing</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPlainTextEdit" name="Alice_TE">
    <property name="geometry">
     <rect>
      <x>13</x>
      <y>70</y>
      <width>161</width>
      <height>51</height>
     </rect>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="Bob_TE">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>70</y>
      <width>161</width>
      <height>51</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="Alice_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Alice</string>
    </property>
   </widget>
   <widget class="QLabel" name="Bob_label">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>50</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Bob</string>
    </property>
   </widget>
   <widget class="QLabel" name="Private_Bob_label">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>210</y>
      <width>231</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Private key (e):</string>
    </property>
   </widget>
   <widget class="QLabel" name="Private_Al_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>210</y>
      <width>241</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>ArrowCursor</cursorShape>
    </property>
    <property name="text">
     <string>Private key (e):</string>
    </property>
   </widget>
   <widget class="QLabel" name="Public_Al_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>190</y>
      <width>241</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Public key (s):</string>
    </property>
   </widget>
   <widget class="QLabel" name="Public_Bob_label">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>190</y>
      <width>241</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Public key (s):</string>
    </property>
   </widget>
   <widget class="QLabel" name="P_Al_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>130</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>P:</string>
    </property>
   </widget>
   <widget class="QLabel" name="Q_Al_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>150</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Q:</string>
    </property>
   </widget>
   <widget class="QLabel" name="N_Al_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>170</y>
      <width>241</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>N:</string>
    </property>
   </widget>
   <widget class="QLabel" name="P_Bob_label">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>130</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>P:</string>
    </property>
   </widget>
   <widget class="QLabel" name="Q_Bob_label">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>150</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Q:</string>
    </property>
   </widget>
   <widget class="QLabel" name="N_Bob_label">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>170</y>
      <width>241</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>N:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Send_C_Bob_button">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>100</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Send [C]</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Generate_keys_button">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>10</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Generate keys</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Send_C_Al_button">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>100</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Send [C]</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Decode_Al_button">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>130</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Decode</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Decode_Bob_button">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>130</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Decode</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Send_Al_button">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>70</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Send</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Send_Bob_button">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>70</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Send</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Clear_button">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>10</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear forms</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>616</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>Send_Al_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendFromAlice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>102</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Send_C_Al_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendCFromAlice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Send_Bob_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendFromBob()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>557</x>
     <y>102</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Send_C_Bob_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendCFromBob()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>557</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Decode_Al_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>decodeAlice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>162</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Decode_Bob_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>decodeBob()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>557</x>
     <y>162</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Clear_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>clearForms()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>340</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Generate_keys_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>generateKeys()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>230</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>generateKeys()</slot>
  <slot>clearForms()</slot>
  <slot>sendFromAlice()</slot>
  <slot>sendFromBob()</slot>
  <slot>sendCFromAlice()</slot>
  <slot>sendCFromBob()</slot>
  <slot>decodeAlice()</slot>
  <slot>decodeBob()</slot>
 </slots>
</ui>
